File: app/src/main/java/com/example/proyeksp/ui/RekeningAdapter.java
package com.example.proyeksp.ui;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.proyeksp.R;
import com.example.proyeksp.database.Rekening;
import com.example.proyeksp.helper.CurrencyHelper;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.time.format.DateTimeFormatter;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class RekeningAdapter extends RecyclerView.Adapter<RekeningAdapter.RekeningViewHolder> {
    private final LayoutInflater mInflater;
    private List<Rekening> rekeningList;
    private final DateFormat dateFormat;
    private final Context context;

    public RekeningAdapter(Context context, List<Rekening> rekenings) {
        this.rekeningList = rekenings;
        this.context = context;

        dateFormat = new SimpleDateFormat("dd MMM yyyy, HH:mm", Locale.getDefault());
        mInflater = LayoutInflater.from(context);
    }

    @NonNull
    @Override
    public RekeningViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = mInflater.inflate(R.layout.setoran_item,parent,false);
        return new RekeningViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RekeningViewHolder holder, int position) {
        Rekening current = rekeningList.get(position);
        long longDate = current.getTglTrans();

        holder.tvNama.setText(current.getNama());
        holder.tvNoRek.setText(current.getNoRek());

        if (longDate == 0) {
            holder.tvTgl.setTextColor(Color.RED);
            holder.tvTgl.setText(context.getResources().getText(R.string.no_date_setor));
            holder.tvSetor.setText("-");
        } else {
            holder.tvSetor.setText(CurrencyHelper.format(current.getSetoran()));
            holder.tvTgl.setTextColor(Color.rgb(34, 177, 76));
            holder.tvTgl.setText(dateFormat.format(new Date(longDate)));
        }
    }

    @Override
    public int getItemCount() {
        return rekeningList.size();
    }

    static class RekeningViewHolder extends RecyclerView.ViewHolder {
        TextView tvNama, tvNoRek, tvSetor, tvTgl;
        public RekeningViewHolder(@NonNull View itemView) {
            super(itemView);

            tvNama = itemView.findViewById(R.id.setor_item_nama);
            tvNoRek = itemView.findViewById(R.id.setor_item_norek);
            tvSetor = itemView.findViewById(R.id.setor_item_setoran);
            tvTgl = itemView.findViewById(R.id.setor_tgl_transaksi);
        }
    }
}

File: app/src/main/java/com/example/proyeksp/ui/RekeningAdapter.kt
package com.example.proyeksp.ui

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.proyeksp.R
import com.example.proyeksp.database.Rekening
import com.example.proyeksp.helper.CurrencyHelper
import com.example.proyeksp.ui.RekeningAdapter.RekeningViewHolder
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class RekeningAdapter(private val context: Context, private val rekeningList: List<Rekening?>?) :
    RecyclerView.Adapter<RekeningViewHolder>() {
    private val mInflater: LayoutInflater = LayoutInflater.from(context)
    private val dateFormat: DateFormat =
        SimpleDateFormat("dd MMM yyyy, HH:mm", Locale.getDefault())

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RekeningViewHolder {
        val view = mInflater.inflate(R.layout.setoran_item, parent, false)
        return RekeningViewHolder(view)
    }

    override fun onBindViewHolder(holder: RekeningViewHolder, position: Int) {
        val current = rekeningList?.get(position)
        val longDate = current?.tglTrans

        holder.tvNama.text = current?.nama
        holder.tvNoRek.text = current?.noRek

        if (longDate == 0L) {
            holder.tvTgl.setTextColor(Color.RED)
            holder.tvTgl.text = context.resources.getText(R.string.no_date_setor)
            holder.tvSetor.text = "-"
        } else {
            if (current != null) {
                holder.tvSetor.text = CurrencyHelper.format(current.setoran)
            }
            holder.tvTgl.setTextColor(Color.rgb(34, 177, 76))
            holder.tvTgl.text = dateFormat.format(longDate?.let { Date(it) })
        }
    }

    override fun getItemCount(): Int {
        return rekeningList!!.size
    }

    class RekeningViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvNama: TextView = itemView.findViewById(R.id.setor_item_nama)
        var tvNoRek: TextView = itemView.findViewById(R.id.setor_item_norek)
        var tvSetor: TextView = itemView.findViewById(R.id.setor_item_setoran)
        var tvTgl: TextView = itemView.findViewById(R.id.setor_tgl_transaksi)
    }
}
